ClientCore/ClientConfiguration.cs

```diff
@@ -569,6 +569,16 @@ public bool DisplayPlayerCountInTopBar
         /// </summary>
         public string LauncherExe => clientDefinitionsIni.GetStringValue(SETTINGS, "LauncherExe", string.Empty);
 
+        public bool EnableRandomMapButton
```

Comment:
Instead of this key in ClientDefinitions.ini, I wonder if it wouldn't be better to just always add the Random Map button to the lobby, and just disable it by default. Then it could be enabled in MultiplayerGameLobby.ini like this

```
[btnPickRandomMap]
Enabled=true
Visible=true
```

That'd get rid of a key in ClientDefinitions.ini and also get rid of multiple `if` conditions in the code, simplifying it. The downside would be that enabling the button would be slightly less obvious.

Reply:
The random map button also gets hidden together with the map list and search bar in multiplayer game lobbies depending on if the player is host or not and when you use the appropriate chat command. If it was implemented like you suggested, there'd be no way for the client to tell what the initial state of the map button is and as a result it would show up even if it wasn't enabled in INI.